inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/TranslateFunction.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.util.List;
+
+public class TranslateFunction implements ValueParser {
+
+    private ValueParser originalStrParser;
+
+    private ValueParser findCharsParser;
+
+    private ValueParser replaceCharsParser;
+
+    public TranslateFunction(Function expr) {
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        originalStrParser = OperatorTools.buildParser(expressions.get(0));
+        findCharsParser = OperatorTools.buildParser(expressions.get(1));
+        replaceCharsParser = OperatorTools.buildParser(expressions.get(2));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object originalStrObject = originalStrParser.parse(sourceData, rowIndex, context);
+        Object findCharsObject = findCharsParser.parse(sourceData, rowIndex, context);
+        Object replaceCharsObject = replaceCharsParser.parse(sourceData, rowIndex, context);
+        String originalStr = OperatorTools.parseString(originalStrObject);
+        String findChars = OperatorTools.parseString(findCharsObject);
+        String replaceChars = OperatorTools.parseString(replaceCharsObject);
+
+        if (originalStr == null) {
+            return "";
+        }
+        StringBuilder builder = null;
+        int replaceSize = replaceChars == null ? 0 : replaceChars.length();
+        for (int i = 0, size = originalStr.length(); i < size; i++) {
+            char ch = originalStr.charAt(i);
+            int index = findChars.indexOf(ch);
```

Comment:
It is better to create a Map in advance to map the relationship between the find character and the replace character to avoid calling the `indexOf` function in the for loop. The time complexity of the indexOf function is `O(n)`

Reply:
done, PTAL!